### PR TITLE
auto tb on vvv+ or debug

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

Gather more debug info in case module dev did not

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

module_utils/basic

##### ANSIBLE VERSION

```
2.4
```

### CODE REVIEW

**Comment by alikins** [comment]
What kind of scenario would cause fail_json to be called without exception info?

Just programmer error/choice when handling an exception? ie, something like

``` python

try:
   37 / 0
except ZeroDivisionError:
   self.fail_json(msg='math is hard')

```

File: lib/ansible/module_utils/basic.py [hunk 1]
```diff
@@ -2134,6 +2134,11 @@ def fail_json(self, **kwargs):
         if 'changed' not in kwargs:
             kwargs['changed'] = False
 
+        # add traceback if debug or high verbosity and it is missing
+        # note: badly named as exception, it is really always been 'traceback'
+        if 'exception' not in kwargs and (self._debug or self._verbosity >= 3):
```
Comment (abadger) [inline lib/ansible/module_utils/basic.py]: I'd add a check that there's a traceback being processed:
``` python
if 'exception' not in kwargs and sys.exc_info()[2] and (self._debug or self._verbosity >= 3):
```

That way the handling of results['exception'] in controller-side code will be easier.

**Comment by bcoca** [comment]
@alikins mostly, yes, if you look most fail_json do not pass traceback info, in most cases it is not needed, but it is very nice to have when debug or high verbosity is involved.

@abadger good catch, will fix

**Comment by abadger** [comment]
+1 to merge from me.

**Comment by dagwieers** [comment]
I'd like to reconsider this change. It is confusing and misguiding as in most cases a module failure (when called by **fail_json**) did not fail because of an exception, and we include the last raised (but caught) exception which is unrelated to the failure.

If anything, this change will cause people to *not* add a traceback when it *is* useful because it is now done automatically. Which is a real anti-pattern.

**Pull request merged**